test/common/algorithm.h

```diff
@@ -25,6 +25,26 @@ class AlgorithmTestSuite : public CxxTest::TestSuite {
 		return true;
 	}
 
+	template<typename It>
+	bool checkEqual(It one_first, It one_last, It other_first) {
+		if (one_first == one_last) 
+			return true;
+
+		// Check whether two containers have the same items in the same order,
+		// starting from some iterators one_first and other_first
+		// 
+		// It iterates through the containers, comparing the elements one by one.
+		// If it finds a discrepancy, it returns false. Otherwise, it returns true.
+
+		for (; one_first != one_last; ++one_first) {
+			if (*one_first != *other_first) {				
+				return false;
+			} 
+			++other_first;
```

Comment:
Note in case you didn't know (and I saw it in str.cpp also but didn't comment): you can have both variable increment in the for.
for (; one_first != one_last; ++one_first, ++other_first) {